src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -0,0 +1,200 @@
+                .id("request1")
+                .imp(Collections.singletonList(imp))
+                .site(site)
+                .build();
+
+        // bid response
+        final Bid bid = Bid.builder()
+                .id("bid1")
+                .impid("imp1")
+                .price(BigDecimal.valueOf(1.5))
+                .adm("<div>Ad Markup</div>")
+                .build();
+
+        final SeatBid seatBidWithBid = SeatBid.builder()
+                .bid(Collections.singletonList(bid))
+                .seat("seat1")
+                .build();
+
+        final BidResponse bidResponse = BidResponse.builder()
+                .id("response1")
+                .seatbid(Collections.singletonList(seatBidWithBid))
+                .cur("USD")
+                .build();
+
+        final BidRejectionTracker bidRejectionTracker = new BidRejectionTracker(
+                "seat2",
+                Set.of("imp1"),
+                1.0);
+
+        bidRejectionTracker.reject("imp1", BidRejectionReason.NO_BID);
+
+        return AuctionContext.builder()
+                .httpRequest(HttpRequestContext.builder().build())
+                .bidRequest(bidRequest)
+                .bidResponse(bidResponse)
+                .bidRejectionTrackers(
+                        Collections.singletonMap(
+                                "seat2",
+                                bidRejectionTracker))
+                .build();
+    }
+
+    public static AuctionContext setUpAuctionContextWithNoBidResponse() {
+        // bid request
+        final Site site = Site.builder()
+                .domain("www.leparisien.fr")
+                .build();
+
+        final BidRequest bidRequest = BidRequest.builder()
+                .id("request1")
+                .site(site)
+                .build();
+
+        return AuctionContext.builder()
+                .httpRequest(HttpRequestContext.builder().build())
+                .bidRequest(bidRequest)
+                .build();
+    }
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        final ObjectMapper mapper = ObjectMapperProvider.mapper();
+
+        final JacksonMapper jacksonMapper = new JacksonMapper(mapper);
+
+        final GreenbidsAnalyticsProperties greenbidsAnalyticsProperties = GreenbidsAnalyticsProperties.builder()
+                .pbuid("pbuid1")
+                .greenbidsSampling(1.0)
+                .exploratorySamplingSplit(0.9)
+                .analyticsServerVersion("2.2.0")
+                .analyticsServer("http://localhost:8080")
+                .configurationRefreshDelayMs(10000L)
+                .timeoutMs(100000L)
+                .build();
+
+        greenbidsAnalyticsReporter = new GreenbidsAnalyticsReporter(
+                greenbidsAnalyticsProperties,
+                jacksonMapper,
+                httpClient);
+    }
+
+    @Test
+    public void shouldReceiveValidResponseOnAuctionContext() {
```

Comment:
all the different code branches and the failed outcomes should be covered by tests

please add more tests

Reply:
added more tests
```
shouldReceiveValidResponseOnAuctionContextForBanner
shouldReceiveValidResponseOnAuctionContextForVideo
shouldFailWhenBidResponseIsNull
shouldFailWhenAuctionContextIsNull
shouldFailOnEncodeException
shouldFailOnUnexpectedResponseStatus
shouldFailWhenAdUnitsListIsEmpty
shouldFailWhenBannerFormatIsNull
shouldFailOnDecodingImpExtPrebid
```